code/modules/reagents/chemistry/reagents/cat2_medicine_reagents.dm

```diff
@@ -431,21 +431,60 @@
 /*Suffix: -rite*/
 /datum/reagent/medicine/C2/penthrite
 	name = "Penthrite"
-	description = "An explosive compound used to stabilize heart conditions. May interfere with stomach acid!"
+	description = "An expensive medicine that aids with pumping blood around the body even without a heart, and prevents the heart from slowing down. It reacts violently with other emergency medication."
 	color = "#F5F5F5"
-	self_consuming = TRUE
+	overdose_threshold = 50
 
 /datum/reagent/medicine/C2/penthrite/on_mob_add(mob/living/M)
 	. = ..()
+	to_chat(M,"<span class='notice'>Your heart begins to beat with great force!")
 	ADD_TRAIT(M, TRAIT_STABLEHEART, type)
+	ADD_TRAIT(M, TRAIT_NOHARDCRIT,type)
+	ADD_TRAIT(M, TRAIT_NOSOFTCRIT,type)
+	M.crit_threshold = M.crit_threshold + HEALTH_THRESHOLD_FULLCRIT*2 //your heart is still pumping!
 
-/datum/reagent/medicine/C2/penthrite/on_mob_metabolize(mob/living/M)
+
+/datum/reagent/medicine/C2/penthrite/on_mob_life(mob/living/carbon/human/H)
+	H.adjustOrganLoss(ORGAN_SLOT_STOMACH,0.25)
+	if(H.health <= HEALTH_THRESHOLD_CRIT && H.health > H.crit_threshold) //we cannot save someone above our raised crit threshold.
+
+		H.adjustToxLoss(-2 * REM, 0)
+		H.adjustBruteLoss(-2 * REM, 0)
+		H.adjustFireLoss(-2 * REM, 0)
+		H.adjustOxyLoss(-6 * REM, 0)
+
+		H.losebreath = 0
+
+		H.adjustOrganLoss(ORGAN_SLOT_HEART,max(1,volume/10)) // your heart is barely keeping up!
+
+		H.Jitter(rand(0,2))
+		H.Dizzy(rand(0,2))
+
+
+		if(prob(33))
+			to_chat(H,"<span class='danger'>Your body is trying to give up, but your heart is still beating!")
+
+	if(H.health <= H.crit_threshold) //certain death above this threshold
+		REMOVE_TRAIT(H, TRAIT_STABLEHEART, type) //we have to remove the stable heart before we give him heart attack
+		to_chat(H,"<span class='danger'>You feel something rupturing inside your chest!")
+		H.emote("scream")
+		H.set_heartattack(TRUE)
+		volume = 0
 	. = ..()
-	M.adjustOrganLoss(ORGAN_SLOT_STOMACH,0.5 * REM)
 
 /datum/reagent/medicine/C2/penthrite/on_mob_end_metabolize(mob/living/M)
+	M.crit_threshold = M.crit_threshold - HEALTH_THRESHOLD_FULLCRIT*2 //your heart is still pumping!
 	REMOVE_TRAIT(M, TRAIT_STABLEHEART, type)
+	REMOVE_TRAIT(M, TRAIT_NOHARDCRIT,type)
+	REMOVE_TRAIT(M, TRAIT_NOSOFTCRIT,type)
 	. = ..()
 
+/datum/reagent/medicine/C2/penthrite/overdose_process(mob/living/carbon/human/H)
+	H.adjustStaminaLoss(10)
+	H.adjustOrganLoss(ORGAN_SLOT_HEART,10)
+	H.set_heartattack(TRUE)
+	H.reagents?.chem_temp +=(-10*REM) // your heart stopped pumping. You are getting colder dear friend
```

Comment:
remove this temperature alteration part of the OD (or make it a part of heart attack code, but I think that it'd still be out of place there)